src/js/Router.js

```diff
@@ -88,9 +106,9 @@ export default class Router extends String {
         if ([null, undefined].includes(params) || !match) return match;
 
         const routeObject = new Route(current, route, this._config);
+        const routeParams = JSON.parse(JSON.stringify({...rParams, ...query})); // Remove undefined params
```

Comment:
```suggestion
        const routeParams = JSON.parse(JSON.stringify(matchedParams)); // Remove undefined params
```

Also, is removing the undefined params actually necessary here?

Reply:
Test `same(route().current('pages.optionalExtension', ''), true);` fails without it